src/pages/projectSettings/tabs/GeneralTab/Fields/RepoConfigField.tsx

```diff
@@ -118,7 +118,11 @@ export const AttachDetachModal: React.FC<
         `There was an error attaching the project: ${err.message}`
       );
     },
-    refetchQueries: ["ProjectSettings", "RepoSettings"],
+    refetchQueries: [
+      "ProjectSettings",
+      "RepoSettings",
+      "GetViewableProjectRefs",
```

Comment:
We want to ensure that the project dropdown correctly reflects whether a project is attached to a repo, so refetch that data when this setting is changed